components/analytics/bot-performance-chart.tsx

```
@@ -0,0 +1,373 @@
+  // Make sure dailyStats are sorted by date
+  const sortedStats = [...dailyStats].sort((a, b) =>
+    dayjs(a.date).isBefore(dayjs(b.date)) ? -1 : 1
+  )
+
+  // Extract all unique platforms
+  const allPlatforms = sortedStats.reduce((platforms, stat) => {
+    Object.keys(stat.platforms).forEach(platform => {
+      if (!platforms.includes(platform)) {
+        platforms.push(platform)
+      }
+    })
```

Biome (1.9.4)

[error] Prefer for...of instead of forEach.

forEach may lead to performance issues when working with large arrays. When combined with functions like filter or map, this causes multiple iterations over the same type.

(lint/complexity/noForEach)